src/rules/no-extraneous-dependencies.js

```diff
@@ -131,10 +115,15 @@ module.exports = {
     ],
   },
 
-  create: function (context) {
+  create(context) {
     const options = context.options[0] || {}
     const filename = context.getFilename()
-    const deps = getDependencies(context, options.packageDir)
+    const deps = packageLocator.readUpSync(
```

Comment:
it seems kind of odd to instantiate `packageLocator` only once and then use it only once to get at the `readUpSync` instance method. Perhaps `readUpSync` could just be exported directly as a singleton function, or perhaps `CachedPackageLocator` could be a factory function instead?